src/web/test/helpers.test.ts

```diff
@@ -0,0 +1,149 @@
+import * as helpers from '../helpers';
+
+describe('getRootPathFromRunnablePath', () => {
+  const root = '/root';
+  it('should return root path when fullPath includes "/u/"', () => {
+    const fullPath = '/root/u/runnable';
+    expect(helpers.getRootPathFromRunnablePath(fullPath)).toBe(root);
+  });
+
+  it('should return root path when fullPath includes "/f/"', () => {
+    const fullPath = '/root/f/runnable';
+    expect(helpers.getRootPathFromRunnablePath(fullPath)).toBe(root);
+  });
+
+  it('should return root path when fullPath ends with "/u/"', () => {
+    const fullPath = '/root/u/';
+    expect(helpers.getRootPathFromRunnablePath(fullPath)).toBe(root);
+  });
+
+  it('should return root path when fullPath ends with "/f/"', () => {
+    const fullPath = '/root/f/';
+    expect(helpers.getRootPathFromRunnablePath(fullPath)).toBe(root);
+  });
+
+  it('should return undefined when fullPath does not include or end with "/u/" or "/f/"', () => {
+    const fullPath = '/root/runnable';
+    expect(helpers.getRootPathFromRunnablePath(fullPath)).toBeUndefined();
+  });
+});
+
+
+describe('determineLanguage', () => {
+  it('should return "python3" when extension is "py"', () => {
+    const path = 'path/to/file.py';
+    const expected = 'python3';
+    expect(helpers.determineLanguage(path)).toBe(expected);
+  });
+
+  it('should return "go" when extension is "go"', () => {
+    const path = 'path/to/file.go';
+    const expected = 'go';
+    expect(helpers.determineLanguage(path)).toBe(expected);
+  });
+
+  it('should return "bash" when extension is "sh"', () => {
+    const path = 'path/to/file.sh';
+    const expected = 'bash';
+    expect(helpers.determineLanguage(path)).toBe(expected);
+  });
+
+  it('should return "graphql" when extension is "gql"', () => {
+    const path = 'path/to/file.gql';
+    const expected = 'graphql';
+    expect(helpers.determineLanguage(path)).toBe(expected);
+  });
+
+  it('should return "powershell" when extension is "ps1"', () => {
+    const path = 'path/to/file.ps1';
+    const expected = 'powershell';
+    expect(helpers.determineLanguage(path)).toBe(expected);
+  });
+
+  // FIXME: I noticed that this was behaving unexpectedly in both the original code and the new refactored code.
```

Comment:
@rubenfiszel Wasn't sure how you wanted to go about this. I noticed this when I was getting the unit tests done.

Reply:
flow should be under `X.flow/flow.yaml` not `/flow/flow.yaml`

Reply:
Awesome! Thank you will fix! I haven't had any experience with flows so this helps!